product_uom/data/product_uom.xml

```diff
@@ -84,5 +84,16 @@
         <field name="category_id" ref="product_uom_category_time"/>
         <field name="uom_type">bigger</field>
     </record>
-
+    <!-- Temperature -->
+    <record id="product_uom_celsius" model="product.uom">
+        <field name="name">Celsius</field>
+        <field name="category_id" ref="product_uom_category_temp"/>
+        <field name="uom_type">reference</field>
+    </record>
+    <record id="product_uom_fahrenheit" model="product.uom">
+        <field name="name">Fahrenheit</field>
+        <field name="factor_inv" eval="33.8"/>
```

Comment:
@jimmyec The conversion is T(°F) = T(°C) × 1.8 + 32

and you are saying that T(°F) = T(°C) × 33.8. It's only true for 1 degree Celsius.

I suggest you remove the Fahrenheit uom or you customize uom to support the correct formula.